### PR TITLE
use `giterr_set_str()` wherever possible

### PR DESCRIPTION
**Pull request by pranitbauva1997**
`giterr_set()` is used when it is required to format a string, and since
we don't really require it for this case, it is better to stick to
`giterr_set_str()`.

This also suppresses a warning(-Wformat-security) raised by the compiler.

### CODE REVIEW

**Comment by ethomson** [comment]
I'm not sure that I fully understand your comment - `create_index_error` does pass a `const char *` for example.  That aside, this is a good fix since treating strings that contain an error message as a format string is definitely incorrect.  (If the error message was `%s`, obviously this would be a bummer.)

**Comment by carlosmn** [comment]
What compiler are you using? We do run with `-Wall` by default, and I haven't seen such warnings. As a general rule, C does not complain when you _add_ `const`ness to a variable.

In the places were we don't expect to perform any substitution, we should rather be using `giterr_set_str()` which doesn't involve any parsing machinery.

File: src/index.c [hunk 1]
```diff
@@ -2358,7 +2358,7 @@ static size_t read_entry(
 		entry.path = (char *)path_ptr;
 	} else {
 		size_t varint_len;
-		size_t shared = git_decode_varint((const unsigned char *)path_ptr, 
+		size_t shared = git_decode_varint((const unsigned char *)path_ptr,
```
Comment (carlosmn) [inline src/index.c]: Let's not do unrelated whitespace changes.

Comment (pranitbauva1997) [inline src/index.c]: Sure! My editor corrected it by default.

**Comment by pranitbauva1997** [comment]
@ethomson I was unsure about it. But since that warning only occurs when the above-specified thing happens, I _assumed_ it to be some technicality within variable arguments (I am not sure about this). I guess it is better to remove it altogether.

**Comment by pranitbauva1997** [comment]
@carlosmn There seems to be a problem with both of our local builds I guess because after I pushed this out, I was checking out the travis-ci builds for [master](https://travis-ci.org/libgit2/libgit2/jobs/176126555) as well as for this [PR](https://travis-ci.org/libgit2/libgit2/builds/176476155) where you can easily see that it is displaying 3 warnings in which this is one of them.
I did not add an extra "-Wall flag".

Also, I am not getting the other two warnings locally which are clearly shown in the master build on travis-ci.

**Comment by pranitbauva1997** [comment]
@ethomson @carlosmn I think using `giterr_set_str` would be a better way out! Thanks!

**Comment by pranitbauva1997** [comment]
@carlosmn @ethomson I have updated the PR.

**Comment by carlosmn** [comment]
The warning I see on that build is

    /home/travis/build/libgit2/libgit2/src/index.c:555:2: warning: format not a string literal and no format arguments [-Wformat-security]

 which is the issue with formatting that @ethomson mentioned, it's nothing to do with `const`.

Ubuntu used to harden their GCC and the version string on Travis suggests they're still adding patches which is why I don't see this on my Debian GCC.

**Comment by carlosmn** [comment]
The warning is likely a result of #4002 which told the compiler that we accept format strings. @pks-t fixed a lot of warnings, but presumably his GCC doesn't contain `-Wformat-security` in the "all" flags either.

**Comment by pranitbauva1997** [comment]
@carlosmn Thanks for explaining! I was under the impression that it would only occur because of `const` things.

**Comment by pks-t** [comment]
I've got a hardened gcc, as well, but it in fact does not contain "-Wformat-security".

Which makes me think - how about adding a build job that compiles with "-Werror" so that we catch compilation warnings early on? Oh wait, there are these curl warnings which would cause the build to fail... sucks :/

**Comment by carlosmn** [comment]
That curl weirdness is disappointing, but I'd like to get rid of curl regardless. We can certainly add a few choice variables like `-Wformat-security` to the block of add-if-supported like we do for `-Wdocumentation`.

**Comment by pks-t** [comment]
Makes sense to me, yes
